xwayland/server.c

```diff
@@ -219,7 +220,8 @@ static void handle_display_destroy(struct wl_listener *listener, void *data) {
 	wlr_xwayland_server_destroy(server);
 }
 
-static int xserver_handle_ready(int signal_number, void *data) {
+static int xserver_handle_ready(int fd, uint32_t mask, void *data) {
```

Comment:
I think it'd be better to check for the mask here. If it contains `WL_EVENT_READABLE`, then we're good. If it contains `WL_EVENT_HANGUP`, then Xwayland startup failed.

Reply:
Considering the race I described above, we can't rely on seeing a WL_EVENT_HANGUP. Yeah, we could return early if we see one, but not seeing one won't imply that there was no failure. IOW, handling WL_EVENT_HANGUP won't remove any of the existing code in the function, it's superfluous.

Reply:
OK